File: rock-paper-scissors-javascript/src/game.js
import { Utility } from "./utils.js";
export class RockPaperScissors {
    constructor() {
        this.util = new Utility();
        this.system_choice = this.util.getSystemChoice();
    }

    play() {
        var user_choice = this.util.getUserInput();
        alert(`System ${this.system_choice}, You ${user_choice}`);
        if (this.util.checkWinCondition(user_choice)) {
            return 1;
        } else if (this.util.checkLoseCondition(user_choice)) {
            return 0;
        } else {
            return 2;
        }
    }
}

File: rock-paper-scissors-python/src/game.py
from utils import Utility


class RockPaperScissors:
    def __init__(self):
        self.util = Utility()
        self.system_choice = self.util.getSystemChoice()

    def play(self):
        user_choice = self.util.getUserChoice()
        print(f"System: {self.system_choice}, You: {user_choice}")
        if self.util.checkWinCondition(user_choice.upper(), self.system_choice):
            print(f"You Win")
        elif self.util.checkLoseCondition(
            user_choice.upper(), self.system_choice
        ):
            print(f"You Lose")
        else:
            print(f"It's a draw")
